tensorflow/compiler/tf2xla/xla_compiler.h

```diff
@@ -192,7 +192,7 @@ class XlaCompiler {
     // here, but on some devices (notably, GPUs), TensorFlow tends to eagerly
     // allocate most or all available memory on the device, leaving none for the
     // compiler to access, unless it can use TensorFlow's allocator.
-    se::DeviceMemoryAllocator* device_allocator = nullptr;
+    std::shared_ptr<se::DeviceMemoryAllocator> device_allocator;
```

Comment:
Please add some breadcrumbs on why this is a `shared_ptr` (they're otherwise unusual).